### DISCUSSION TITLE
[Compilers]<Disabling errors/warning in zksolc>

### DISCUSSION THREAD

**Discussion by Szczepoo13** [zkSync-Community-Hub/zksync-developers#740] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.20

### zksolc/zkvyper version?

1.5.6

### Explain the issue

Hey,

I'm trying to compile a contract and encountering the following error:
![reentrancy](https://github.com/user-attachments/assets/7892297b-9f88-45ad-b14f-32cf2d0a6600)

As mentioned in the error message, it's possible to disable the error by:

**1)** As I understand, manually creating a JSON input file with suppressedErrors = ["sendtransfer"] when compiling the code manually with zkSolc in Standard JSON Mode. However, when using tools like HardHat or Foundry (e.g., running forge build), the JSON input is dynamically generated by the tool. This makes it difficult to access or modify the input file directly (reference: https://docs.zksync.io/zk-stack/components/compiler/toolchain/solidity#standard-json). 
Is there any other way to edit or include input JSON for the compiler while using Foundry-zksync for example?

Additionally, I have a question when using foundry for deployment:

Does Foundry-zksync provide or enable users to ignore error codes in the **Foundry.toml** file?? For example, in the original Foundry, we can use something like:  **_ignored_warnings_from = ["src/periphery/contracts/treasury/RevenueSplitter.sol"]_** 
I've done some testing, and it seems like these functionalities are not enabled or working as expected with Foundry-zksync. Could you confirm whether this feature is supported, or if there's an alternative way to ignore specific errors or warnings during deployment? :)

I have tested a list of supported aliases for the original Foundry:
[(https://github.com/foundry-rs/foundry/blob/00cec1d2445ad9b8917a377cb6f9b755728eb9da/crates/config/src/error.rs#L202-L214)
](https://github.com/foundry-rs/foundry/blob/00cec1d2445ad9b8917a377cb6f9b755728eb9da/crates/config/src/error.rs#L202-L214)

**2)** 
It also appears like **--suppress-errors sendTransfer** is not supported by Foundry-zksync, I have come across the flag:
 **"forge build --ignored-error-codes <codes>"** but this one will not be enough since it silences only warnings.

**Extra)**
Therefore, If both 1) (modifying the JSON input) and 2) (suppressing errors via the CLI) are not possible, am I correct in assuming that the only remaining solution to bypass the compilation error would be refactoring the code?

**Comment by alhenaai** [comment]
Hello,

Regarding your query about disabling errors or warnings in `zksolc` when using Foundry-zksync, the current documentation and tools do not explicitly support the direct modification of the JSON input for the compiler or the suppression of specific error codes through configuration files like `Foundry.toml`.

1. **Modifying JSON Input for Compiler**:
   - When using Foundry-zksync, the JSON input is indeed dynamically generated, and there isn't a straightforward method provided in the documentation to manually edit or include specific configurations like `suppressedErrors` directly within this environment.

2. **Ignoring Error Codes in Foundry.toml**:
   - The functionality to ignore specific warnings or errors via `Foundry.toml` similar to what is available in the original Foundry does not appear to be supported in Foundry-zksync. This limitation affects the ability to use configurations like `ignored_warnings_from` to manage compilation behaviors.

3. **Alternative Solutions**:
   - Without the ability to suppress specific errors or modify the JSON input directly, the alternative would indeed involve refactoring the code to avoid triggering the errors or warnings that you are currently facing.

Unfortunately, if these errors are critical and cannot be refactored out, this might limit the use of Foundry-zksync for your specific needs at this time. You might consider reaching out to the ZKsync development community or checking for updates in the toolchain that might address these limitations in the future.

For the most current and detailed guidance, you may want to keep an eye on updates in the ZKsync documentation or community discussions.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
In Hardhat, you can add this in the zksolc settings in the `hardhat.config.ts` file [as detailed in the documentation](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-solc#configuration)

```
import "@matterlabs/hardhat-zksync";

const config: HardhatUserConfig = {

zksolc:{
    settings:{
      suppressedErrors:["sendtransfer"]
    }
  },
```

Let me check what's the corresponding setting in foundry

**Reply by uF4No** [reply]
For foundry-zksync [I've created this feature request](https://github.com/matter-labs/foundry-zksync/issues/633). Please check for updates there.

**Reply by Szczepoo13** [reply]
As always, thank you for your replies @uF4No :)
